tests/unit/test_workflows.py

```diff
@@ -49,5 +57,139 @@ def test_referenced_schemas_exist(self):
+                "required": True,
+                "schemas": [
+                    {
+                        "name": "study",
+                        "required": True,
+                        "allowMultipleObjects": False,
+                    },
+                    {
+                        "name": "dac",
+                        "required": True,
+                        "allowMultipleObjects": True,
+                    },
+                    {
+                        "name": "policy",
+                        "required": False,
+                        "allowMultipleObjects": True,
+                    },
+                ],
+            },
+            {
+                "title": "with require_or",
+                "required": True,
+                "schemas": [
+                    {
+                        "name": "image",
+                        "required": True,
+                        "requires_or": ["dac", "policy"],
+                        "allowMultipleObjects": True,
+                    },
+                ],
+            },
+            {
+                "title": "step not required",
+                "required": False,
+                "schemas": [
+                    {
+                        "name": "bpdataset",
+                        "required": True,
+                        "allowMultipleObjects": False,
+                    },
+                    {
+                        "name": "experiment",
+                        "required": False,
+                        "allowMultipleObjects": True,
+                    },
+                    {"name": "run", "required": False, "allowMultipleObjects": True, "requires": ["experiment"]},
+                ],
+            },
+        ],
+        "publish": [{"name": "datacite", "endpoint": "datacite"}],
+    }
+    WORKFLOWS = {
+        "test": Workflow(valid_workflow_dict),
+        "fail": Workflow(
+            {
+                "name": "fail",
+                "description": "",
+                "steps": [
+                    {
+                        "title": "single step",
+                        "schemas": [
+                            {
+                                "name": "study",
+                                "required": True,
+                            },
+                            {
+                                "name": "dac",
+                                "required": True,
+                                "allowMultipleObjects": True,
+                            },
+                            {
+                                "name": "policy",
+                                "required": True,
+                                "allowMultipleObjects": True,
+                            },
+                        ],
+                    }
+                ],
+                "publish": [{"name": "datacite", "endpoint": "datacite", "requiredSchemas": ["datacite"]}],
+            }
+        ),
+    }
```

Comment:
Perhaps these into a test file instead? It can be here too, I don't mind that much

Reply:
Done